File: Python/13-Matrizes/Matriz.py
#O python não possui variaveis compostas multidimensionais (matrizes)
#No python precisamos de um vetor dentro do outro para ter uma matriz

#Aqui podemos ter uma matriz 3x3
matriz = [[1,3,5],[2,4,6],[3,5,7]]

#Para acessar um elemento em uma posição especifica temos que ter colocar
#a linha e a coluna correnpondente
print("Linha 2 Coluna 1")
print(matriz[2][1]) 

print("==================")

#Mostrara 5 porque o indice de ambos começa em 0

print("Linha 0 Coluna 0")
print(matriz[0][0])

print("==================")

print("Matriz completa")

#Para ver toda a matriz precisamos de um for dentro de outro(um para linha e outra para coluna)
for i in matriz :
    for j in i :
        print(j, end = " ")
    print()

File: JavaScript/14-Matrizes/Matriz.js
//Matrizes em JavaScript (|Vetor de Vetores)
//Criamos uma estrutura similar para trabalhar com esse tipo de dados
//Criamos um vetor dentro de outro
var Grid = [
    ["Supra", "Mustang"],//Temos 1 linha e 2 colunas
    ["Skyline", "GTO"], //Temos outra linha
    ["Audi TT", "Corvette"]
]
//Visualizar o conteudo da matriz
//Vamos exibir o conteudo situado na linha 3 (primeiro []) coluna 1(segundo[]).
//Lembrando que começamos da posição [0] para ambos
document.write("O Quinto colocado é " + Grid[2][0] + "<br/>");
document.write("O Pole Position é " + Grid[0][0] + "<br/>");
document.write("--------------------------------" + "<br/>");
//Para ver todos elementos ao mesmo tempo
document.write(Grid + "<br/>");//Sera exibido tudo em uma linha mas em ordem
document.write("--------------------------------" + "<br/>");
//Para exibir a matriz de forma correta precisamos de dois FOR
//Primeiro vamos exibir em forma de lista porem ordenada
var car = "";
for (l = 0; l <= 2; l++) {//Varrera linha por linha
    for (c = 0; c <= 1; c++) { //Varrera cada coluna (temos 2)
        car += Grid[l][c] + "\n";//Concatenar variavel auxiliar com ordenção entre linhas e colunas e uma quebra de linha
    }
}
alert("Grid \n" + car); //Exibição em forma de popup

//Agora na forma original da matriz (nesse caso 3x2)
var car = "";
for (l = 0; l <= 2; l++) {
    for (c = 0; c <= 1; c++) {
        if (c < 1) {
            car += Grid[l][c] + " \t\t"//Tabulações para espaçar melhor
        }
        else {
            car += Grid[l][c] + "\n"; //Agora vem a quebra de linha(temos 2 elementos na coluna)
        }
    }
}
alert("Grid \n" + car); 